pkg/controller/backup_repository_controller.go

```diff
@@ -194,8 +204,14 @@ func (r *BackupRepoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	bsl, bslErr := r.getBSL(ctx, backupRepo)
```

Comment:
It's better we get a `*velerov1api.BackupStorageLocation` (pointer) from `getBSL` and pass the pointer all the way to `checkNotReadyRepo` and `getIdentiferByBSL`, etc.

Reply:
Modified.